libs/common/src/components/modals/signature-modal/confirm-messages.tsx

```diff
@@ -55,6 +55,78 @@ export const ConfirmMessages = observer<ConfirmMessagesProps>(
     const intl = useIntl();
     const [selectedTab, setSelectedTab] = useState(Tab.TransactionDetails);
     const theme = useTheme();
+    const renderButtons = () => {
+      if (props.isLogin) {
```

Comment:
You have four `props.isLogin` checks in this component. I guess that splitting up the component makes things more readable and easier to change the two independent use cases in the future.